### PR TITLE
Remove workbook calls

### PR DESCRIPTION
**Pull request by Kaixhin**
Rather than having to comment out the workbook calls, I've wrapped them in if statements by using `pcall` to work out whether `'lab-workbook'` is available or not. Obviously I can't test if it is available, but now it can run without crashing.

I've also updated the docs to remove this and also mentioned `--dataRoot` because the docs make it sound like the data path is hardcoded in.

### CODE REVIEW

**Comment by Kaixhin** [comment]
I've also just added the initialisation method that they used in the paper for consistency.

**Comment by gcr** [comment]
This looks great, thank you! I'll merge this PR when my current re-run of the experiments finish (tonight or tomorrow morning). (The current set of results were accidentally sampled on 2k testing images, instead of using all 10k sequentially, so I'm re-running them now)

Your `nninit` package looks really clever, thanks for the pointer! Is Kai Ming's initialization any different from the initialization we're already using here? https://github.com/gcr/torch-residual-networks/blob/master/train-cifar.lua#L84-L94

**Comment by Kaixhin** [comment]
Nope, looks good to me. And I double-checked - I missed out the zero bias - so I've added that in. What I'm not certain about is the initialisation for Batch Normalization - in the original paper they used a Gaussian with a small standard deviation (so 0.002 seems reasonable, but I would be interested in knowing if that value was found somewhere). However, it's not clear in the ResNet paper whether they applied their own initialisation scheme to the BN layers...

In any case I've finished all the initialisations and removed your initialisation loop, so it'll be a little bit faster starting up. But more importantly you can try other initialisation schemes and see if they make an impact (especially the orthogonal one - that's supposedly the best one for deep networks).

**Comment by gcr** [comment]
Just rebased your changes into master. Thanks again!

Note that the new weight init seems to affect performance a wee bit (test error is 0.0875 where it was 0.0829 before), but it might just be noise. Probably not a big deal.

train loss

test error

**Comment by Kaixhin** [comment]
Cool. I've matched the init scheme to copy your parameters, so that should just be noise.

If you're taking suggestions for experiments how about trying different activation functions? I was discussing this paper with some colleagues and we wonder if the fact that the ReLU saturates at 0 is key - in that the residual function can be driven to 0 when the identity is "optimal". If this were true than a sigmoid would perform better than tanh, and ReLUs would perform better than Leaky ReLUs and ELUs (it's hard to control for everything but this would be reasonably indicative). If you've got saved models you could also check how sparse the activations are at the layers computing the residual function.

**Comment by gcr** [comment]
Hm! That's an interesting thought. In the paper, Kai writes that part of the "magic" here might be that creating identity connections is easy for the solver, so using ReLU units that saturate at half their input domain would definitely make that easier.

I think the way the batch normalization layer is used also makes it easy for the solver to create identity connections though, since all you need to do is to set the BN scaling weight parameter to 0 to get an identity connection.

It's worth trying!

In the mean time, here's a copy of the saved model that got 8.29% if you'd like to mess around: https://mjw-xi8mledcnyry.s3.amazonaws.com/experiments/201601141709-AnY56THQt7/model.t7
